### ISSUE TITLE
Thumbnail view not appearing in "Open Displays" on Windows

### ISSUE THREAD

**Issue by lhsego** [tesseradata/trelliscope#72]
On Windows, the thumbnail view of the display in "Open Displays" is blank (at least on Chrome, haven't tested it on Firefox).

**Comment by hafen** [comment]
Interesting... this shouldn't be a platform-specific problem unless it has to do with the generation of the thumbnail.  Can you check to see if the thumbnail exists?  Look for `_vdb_directory_/displays/_group_/_name_/thumb.png`.

Also, have you looked at it on mac/linux too?  What kind of panel function is it?  Thumbnails currently only work for panel functions that can create raster images (base R, ggplot, and lattice).

**Comment by lhsego** [comment]
It's happening with base R.  On mac/linux, the `thumb_small.png` file gets created like this:
![thumb_small](https://cloud.githubusercontent.com/assets/5566573/6138883/40e4848e-b13a-11e4-9d9c-b8edfea05757.png)
But on Windows,  `thumb_small.png` is just a white, square, image:  
![thumb_small](https://cloud.githubusercontent.com/assets/5566573/6138899/66f53fc4-b13a-11e4-80c3-be5475cbbb11.png)

**Comment by hafen** [comment]
Oh strange.  I thought your fix took care of this?

**Comment by lhsego** [comment]
No, my fixes just ensured that vdb files actually got copied from the temp folder to the vdb because `file.rename()`, when applied to a directory, does nothing in Windows.  The files have to be renamed individually.  This is something else.  `thumb_small.png` isn't getting written (or written correctly) in Windows.

**Comment by amwhite** [comment]
This is a known issue with the rasterImage function which is used to create the thumbnail. Basically it doesn't work if you're accessing a windows machine via remote desktop. See issue and potential solution (for windows server) here:
http://r.789695.n4.nabble.com/R-base-package-grid-does-not-output-raster-image-td4672188.html

**Comment by amwhite** [comment]
More information about my previous comment. The problem with rasterImage() and windows remote desktop is mentioned in the documentation here: https://stat.ethz.ch/R-manual/R-devel/library/graphics/html/rasterImage.html. It's important to note that the problem here is the remote desktop connection when creating the display (i.e. calling makeDisplay()) not when viewing the display. 

When remoting from a windows machine to a windows machine, you can set the number of colors to Highest Quality (32 bit) like below and that alleviates the problem:
![capture](https://cloud.githubusercontent.com/assets/5456869/6160548/2094bf66-b219-11e4-8e8a-ded90990e4c6.PNG)

When remoting from mac to windows I tried setting colors to Millions instead of Thousands (Preferences-->Display before connecting to remote machine) but RDC just froze. I'm not sure RDC for Mac is stable/powerful enough to handle this but perhaps others should try.
![screen shot 2015-02-11 at 6 12 16 pm](https://cloud.githubusercontent.com/assets/5456869/6160608/01007450-b21a-11e4-8600-8d55335d5c28.png)

**Comment by lhsego** [comment]
Thanks for figuring this out, Amanda!

**Comment by hafen** [comment]
I wonder if we should use something other than rasterImage.  Originally I just made a scaled-down version of the image but had a hard time doing it reliably.  Also, we don't have a good way to get thumbnails for javascript-based panel functions right now as they are rendered in the browser instead of R.  This is extremely low priority to do anything about in my opinion.

**Comment by hafen** [comment]
Least we could do is make it show "no thumbnail available" when there isn't one.
